diff --git a/Makefile.common b/Makefile.common
index 6fe17d3df2..2dfe336da1 100644
--- a/Makefile.common
+++ b/Makefile.common
@@ -1989,8 +1989,6 @@ ifeq ($(HAVE_7ZIP),1)
    DEFINES += -DHAVE_7ZIP -D_7ZIP_ST
    7ZOBJ = $(DEPS_DIR)/7zip/7zArcIn.o \
            $(DEPS_DIR)/7zip/7zBuf.o \
-           $(DEPS_DIR)/7zip/7zCrc.o \
-           $(DEPS_DIR)/7zip/7zCrcOpt.o \
            $(DEPS_DIR)/7zip/7zDec.o \
            $(DEPS_DIR)/7zip/CpuArch.o \
            $(DEPS_DIR)/7zip/Delta.o \
diff --git a/Makefile.libnx b/Makefile.libnx
index e35310bd1d..8ac6c9c9cb 100644
--- a/Makefile.libnx
+++ b/Makefile.libnx
@@ -133,4 +133,4 @@
 CFLAGS := -g -Wall -O3 -fcommon -ffunction-sections \
-          $(ARCH) $(DEFINES) $(INCLUDE_DIRS) -I$(LIBNX)/include -I$(PORTLIBS)/include/ -include $(LIBNX)/include/switch.h #$(shell $(PORTLIBS)/bin/freetype-config --cflags)
+          $(ARCH) $(DEFINES) $(INCLUDE_DIRS) $(GOOSE_UAM_INC) -I$(LIBNX)/include -I$(PORTLIBS)/include/ -include $(LIBNX)/include/switch.h #$(shell $(PORTLIBS)/bin/freetype-config --cflags)

 CFLAGS += $(INCLUDE) -DSWITCH=1 -DHAVE_LIBNX=1 -DNXLINK=1 -DHAVE_SHADERPIPELINE -DHAVE_ONLINE_UPDATER -DHAVE_UPDATE_ASSETS -DHAVE_UPDATE_CORES -DHAVE_UPDATE_CORE_INFO -DHAVE_STB_FONT #-DHAVE_FREETYPE
@@ -151,7 +151,7 @@ LDFLAGS	=	-specs=$(DEVKITPRO)/libnx/switch.specs $(ARCH) -Wl,--allow-multiple-de
 # add things from Makefile.common
 CFLAGS += $(DEF_FLAGS)
 
-LIBS := -lswresample -lavformat -lavcodec -lavutil -lswscale -lstdc++ -lbz2 -lpng -lz -lnx -lvpx -lopus -llzma -lmbedtls -lmbedx509 -lmbedcrypto -lnx -lm
+LIBS := -lswresample -lavformat -lavcodec -lavutil -lswscale -lstdc++ -lbz2 -lpng -lz -lzstd -ljpeg -lwebp -lcpuinfo -luam -lnx -lvpx -lopus -llzma -lmbedtls -lmbedx509 -lmbedcrypto -lnx -lm
 
 ifeq ($(HAVE_OPENGL), 1)
   LIBS := -lEGL -lglapi -ldrm_nouveau $(LIBS)
@@ -206,6 +206,7 @@ endif
 export OFILES 	:=	$(OBJ)
 
 ifeq ($(strip $(HAVE_STATIC_DUMMY)),)
+  OFILES += goose_excpt.o
   OFILES += libretro_libnx.a
 endif
 
